File: oop/abstractClass/java/Player.java
package com.nps.example;

public class Player extends Character {

  private float critChance;

  public Player(String name, int hp, int strength, int defense, float critChance) {
    super(name, hp, strength, defense);

    this.critChance = critChance;
  }

  public float getCritChance() {
    return this.critChance;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void setCritChance(float critChance) {
    this.critChance = critChance;
  }

  @Override
  public float calculateDmg() {
    return this.strength;
  }
}

File: oop/inheritance2/after/player.py
from character import Character

class Player(Character):
  def __init__(self, name, hp, strength, defense, critChance):
    super().__init__(name, hp, strength, defense)

    self.__critChance = critChance

  def getCritChance(self):
    return self.__critChance

  def setCritChance(self, critChance):
    self.__critChance = critChance

  def calculateDmg(self):
    return super().getStrength()

  def __str__(self):
    return f'{super().__str__()}, critChance={self.__critChance}'
